tools/cooja/java/org/contikios/cooja/SafeRandom.java

```diff
@@ -36,26 +36,39 @@
  * This ensures that the functions of the random number generator are
  * only called by the the thread initializing a simulation or the
  * simulation thread itself.
- * Rationale: By allowing another thread to use the random number 
- * generator concurrency is intruduced, thus it can not be guaranteed 
- * that simulations are reproduceable.
+ * Rationale: By allowing another thread to use the random number
+ * generator concurrency is introduced, thus it can not be guaranteed
+ * that simulations are reproducible.
  *
  */
 public class SafeRandom extends Random {
   
   Simulation sim = null;
   Thread initThread = null;
+  Boolean simStarted = false;
```

Comment:
The value will be stored as a Boolean object. One affect is that simStarted can not only be set to true or false but also null. But the overhead in this case is negligible. I was just curious.